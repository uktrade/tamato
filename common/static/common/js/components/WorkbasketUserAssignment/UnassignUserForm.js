/* global CSRF_TOKEN:readonly, unassignUsersUrl:readonly */

import React, { useEffect } from "react";
import accessibleAutocomplete from "accessible-autocomplete";
import PropTypes from "prop-types";

function UnassignUserForm({ users }) {
  const elementId = "assignments-select";
  const elementName = "assignments";
  const label = "Unassign user";
  const hint = "Select a user to unassign";

  useEffect(() => {
    const selectElement = document.getElementById(elementId);
    if (selectElement)
      accessibleAutocomplete.enhanceSelectElement({
        autoselect: false,
        defaultValue: "",
        minLength: 2,
        showAllValues: true,
        selectElement: selectElement,
      });
  }, []);

  return (
    <form
      action={unassignUsersUrl}
      method="POST"
      data-testid={"unassign-user-form"}
    >
      <input type="hidden" value={CSRF_TOKEN} name="csrfmiddlewaretoken" />
      <div className="govuk-form-group">
        <label className="govuk-label" htmlFor={elementId}>
          {label}
        </label>
        <div id={`${elementName}-hint`} className="govuk-hint">
          {hint}
        </div>
        <select
          required={true}
          id={elementId}
          name={elementName}
          data-testid="unassign-user-select"
        >
          <option value="">Select a user</option>
          {users.map((user) => (
            <option key={user.pk} value={user.pk}>
              {user.name}
            </option>
          ))}
        </select>
      </div>
      <button
        type="submit"
        className="govuk-button"
        data-prevent-double-click="true"
      >
        Save
      </button>
    </form>
  );
}

UnassignUserForm.propTypes = {
  users: PropTypes.arrayOf(
    PropTypes.shape({
      pk: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
    })
  ),
};

export { UnassignUserForm };
