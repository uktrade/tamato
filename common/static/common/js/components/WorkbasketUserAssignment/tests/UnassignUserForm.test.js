import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react";
import { UnassignUserForm } from '../UnassignUserForm';


const mockUsers = [
  {
    "pk": 1,
    "name": "User One",
  },
  {
    "pk": 2,
    "name": "User Two",
  },
  {
    "pk": 3,
    "name": "User Three",
  },
]

describe(UnassignUserForm, () => {
  const csrfToken = "abc123";
  const unassignUsersUrl = "/workbaskets/1/unassign-users/";

  beforeAll(() => {
    window.CSRF_TOKEN = csrfToken;
    window.unassignUsersUrl = unassignUsersUrl;
  });
  
  afterAll(() => {
    delete window.CSRF_TOKEN;
    delete window.unassignUsersUrl;
  });

  it('renders form', () => {
    const component = renderer.create(
      <UnassignUserForm users={mockUsers}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has unassign users form action url', () => {
    render(
      <UnassignUserForm users={mockUsers}/>
    );

    expect(screen.getByTestId("unassign-user-form")).toHaveAttribute('action', unassignUsersUrl);
  });
})