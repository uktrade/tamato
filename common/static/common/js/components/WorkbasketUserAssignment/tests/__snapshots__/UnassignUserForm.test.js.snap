// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnassignUserForm renders form 1`] = `
<form
  action="/workbaskets/1/unassign-users/"
  data-testid="unassign-user-form"
  method="POST"
>
  <input
    name="csrfmiddlewaretoken"
    type="hidden"
    value="abc123"
  />
  <div
    className="govuk-form-group"
  >
    <label
      className="govuk-label"
      htmlFor="assignees-select"
    >
      Unassign user
    </label>
    <div
      className="govuk-hint"
      id="assignees-hint"
    >
      Select a user to unassign
    </div>
    <select
      data-testid="unassign-user-select"
      id="assignees-select"
      name="assignees"
      required={true}
    >
      <option
        value=""
      >
        Select a user
      </option>
      <option
        value={1}
      >
        User One
      </option>
      <option
        value={2}
      >
        User Two
      </option>
      <option
        value={3}
      >
        User Three
      </option>
    </select>
  </div>
  <button
    className="govuk-button"
    data-prevent-double-click="true"
    type="submit"
  >
    Save
  </button>
</form>
`;
